[Instrumentation.Process] Create src/test project files.

Yun-Ting opened the pull request:
## Changes
Working towards: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/447
Creating project skeleton for process instrumentation.

Kielek commented on test/OpenTelemetry.Instrumentation.Process.Tests/OpenTelemetry.Instrumentation.Process.Tests.csproj:
```diff
@@ -0,0 +1,25 @@
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry.Exporter.InMemory" Version="1.3.0" />
```
Please use other variables from: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/build/Common.nonprod.props (InMemory, vs runner, collector).

Kielek reviewed (commented):
I think that you are missing workflow, like: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/workflows/package-Instrumentation.Runtime.yml

reyang commented on test/OpenTelemetry.Instrumentation.Process.Tests/OpenTelemetry.Instrumentation.Process.Tests.csproj:
```diff
@@ -0,0 +1,25 @@
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="xunit" Version="$(XUnitPkgVer)" />
+    <PackageReference Include="coverlet.collector" Version="3.1.2">
```
Consider https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/7386a192f994fc8d2366c18442d665f1248e98f6/test/OpenTelemetry.Instrumentation.Runtime.Tests/OpenTelemetry.Instrumentation.Runtime.Tests.csproj#L9.

reyang reviewed (approved):
LGTM, consider https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#contributing-a-new-project (either in this PR or a follow up PR).

Yun-Ting commented:
> LGTM, consider https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#contributing-a-new-project (either in this PR or a follow up PR).

Thanks for the link! I've added a project readme and updated the component owners doc.

cijothomas commented on src/OpenTelemetry.Instrumentation.Process/OpenTelemetry.Instrumentation.Process.csproj:
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net462;netcoreapp3.1;net6.0</TargetFrameworks>
```
nit - unless we know for sure we need all these targets, we can start with netstandard2.0.

Yun-Ting replied:
removed others

cijothomas commented on src/OpenTelemetry.Instrumentation.Process/README.md:
````diff
@@ -0,0 +1,39 @@
+```csharp
+using var meterProvider = Sdk.CreateMeterProviderBuilder()
+    .AddProcessInstrumentation()
+    .AddPrometheusExporter()
````
nit: need to mention additional nuget installation for prometheus

Yun-Ting replied:
added installation info

xiang17 commented on src/OpenTelemetry.Instrumentation.Process/README.md:
```diff
@@ -0,0 +1,39 @@
+
+This is an [Instrumentation
+Library](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library),
+which instruments [.NET Process](https://docs.microsoft.com/dotnet) and
+collect telemetry about Process behavior.
```
```suggestion
which instruments [.NET](https://docs.microsoft.com/dotnet) process and
collect telemetry about process behavior.
```

Nit: No need to use upper case for process.

Yun-Ting replied:
updated; thanks!

Yun-Ting commented:
> I think that you are missing workflow, like: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/workflows/package-Instrumentation.Runtime.yml

Thanks for spotting it; I've added the workflow.

utpilla commented on opentelemetry-dotnet-contrib.sln:
```diff
@@ -233,6 +233,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Instrumentati
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Instrumentation.Process", "src\OpenTelemetry.Instrumentation.Process\OpenTelemetry.Instrumentation.Process.csproj", "{F811262D-D78A-4C4A-8A31-FFC458164BF2}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Instrumentation.Process.Tests", "test\OpenTelemetry.Instrumentation.Process.Tests\OpenTelemetry.Instrumentation.Process.Tests.csproj", "{61421ACF-5F90-491B-AFB3-14EF12CCA255}"
+EndProject
```
nit: Update .sln file to add the package workflow.

reyang commented:
![image](https://user-images.githubusercontent.com/17327289/186281616-77925200-0105-4550-a051-daf64d9ac1ed.png)

utpilla commented:
> ![image](https://user-images.githubusercontent.com/17327289/186281616-77925200-0105-4550-a051-daf64d9ac1ed.png)

This test has been failing intermittently in the past few days.
